src/lib.rs

```diff
@@ -597,6 +597,19 @@ pub trait StreamingIteratorMut: StreamingIterator {
     {
         MapDerefMut { it: self, f }
     }
+
+    /// Creates an iterator which flattens nested streaming iterators.
+    #[inline]
+    fn flatten(self) -> Flatten<Self>
+    where
+        Self: Sized,
+        Self::Item: StreamingIterator + Sized,
```

Comment:
This doesn't need to be `Sized` anymore -- I think that was only because you didn't have `J: ?Sized` before, but that's gone now anyway.